src/answers.sql

```diff
@@ -1,15 +1,250 @@
+                    ELSE account_from
+                END AS account_id,
+                CASE
+                    WHEN type = 'TRANSFER' THEN mount
+                    ELSE 0
+                END AS mount
+            FROM movements
+        )
+        UNION ALL
+        (
+            -- accounts starting money
+            SELECT id AS account_id,
+                mount
+            FROM accounts
+        )
+    )
+SELECT am.account_id,
+    ROUND(SUM(am.mount)::NUMERIC, 2) AS money
+FROM accounts_money_movements am
+GROUP BY am.account_id;
+END;
+$$ LANGUAGE PLPGSQL;
 
+--
+-- Solution applying function  get_accounts_final_balance
+--
+SELECT
+	U.ID,
+	U.NAME,
+	SUM(AB.MONEY) ALL_MONEY
+FROM
+	GET_ACCOUNTS_FINAL_BALANCE () AB
+	INNER JOIN ACCOUNTS A ON AB.ACCOUNT_ID = A.ID
+	INNER JOIN USERS U ON U.ID = A.USER_ID
+GROUP BY
+	U.ID,
+	U.NAME
+ORDER BY ALL_MONEY DESC
+LIMIT 3;
+
+--
+--
+--
 -- 5
+--
+DO $$
+DECLARE
+    account_balance_a DOUBLE PRECISION;
+    account_balance_b DOUBLE PRECISION;
+	
+    account_id_a UUID := '3b79e403-c788-495a-a8ca-86ad7643afaf';
+    account_id_b UUID := 'fd244313-36e5-4a17-a27c-f8265bc46590';
+BEGIN
+
+	--a. First, get the ammount for the account `3b79e403-c788-495a-a8ca-86ad7643afaf` and `fd244313-36e5-4a17-a27c-f8265bc46590` after all their movements.
+	
+	SELECT money into account_balance_a
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_a;
+
+	SELECT money into account_balance_b
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_b;
+
+	RAISE NOTICE 'Account balance from % is %', account_id_a, account_balance_a;
+	RAISE NOTICE 'Account balance from % is %', account_id_b, account_balance_b;
+
+	--b. Add a new movement with the information: from: 3b79e403-c788-495a-a8ca-86ad7643afaf make a transfer to fd244313-36e5-4a17-a27c-f8265bc46590 mount: 50.75
+	IF account_balance_a < 50.75 THEN
+		RAISE EXCEPTION 'Account balance is insufficient!';
+	ELSE 
+		INSERT INTO movements  (id, type, account_from, account_to, mount) 
+		VALUES (gen_random_uuid(), 'TRANSFER', account_id_a,account_id_b, 50.75);
+		RAISE NOTICE 'Transfer % USD from % to % MADE SUCCESSFULLY!', 50.75, account_id_a, account_id_b;
+	END IF;
+
+	--c. Add a new movement with the information: from: 3b79e403-c788-495a-a8ca-86ad7643afaf type: OUT mount: 731823.56
+
+	SELECT money into account_balance_a
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_a;
+	
+	IF account_balance_a < 500 THEN
+		RAISE EXCEPTION 'Account balance is insufficient!';
+	ELSE 
+		INSERT INTO movements  (id, type, account_from, account_to, mount) 
+		VALUES (gen_random_uuid(), 'OUT', account_id_a,account_id_b , 500);
+		RAISE NOTICE 'Movement OUT % USD from % MADE SUCCESSFULLY!', 500, account_id_a;
+	END IF;
+
+	--f. Once the transaction is correct, make a commit
+
+	SELECT money into account_balance_b
+	FROM get_accounts_final_balance() 
+	WHERE account_id = account_id_b;
+
```

Comment:
Where is the commit?

Reply:
Not need due to the sintaxis I'm using.

```sql

DO $$
DECLARE
--
BEGIN

--

EXCEPTION
--
END;
$$;
```

It treats the whole block as a transaction and only commit once its done